src/testing/cluster.zig

```diff
@@ -219,7 +220,8 @@ pub fn ClusterType(comptime StateMachineType: anytype) type {
             errdefer allocator.free(replica_pools);
 
             const pipeline_requests_limit =
-                options.client_count -| constants.pipeline_prepare_queue_max;
+                @min(options.client_count, constants.clients_max) -|
```

Comment:
I don't get this `@min`, shouldn't we just assert that `client_count` is not greater than clients_max?

EDIT: got it: clients_max is how many clients can be registered with a cluster at the same time. But of course there could be arbitrary number of clients! 

Worth a comment in the code though!